src/sys/fuchsia/handles.rs

```diff
@@ -0,0 +1,80 @@
+use {io, poll, Evented, Ready, Poll, PollOpt, Token};
+use magenta::HandleBase;
+use std::sync::Mutex;
+
+/// Wrapper for registering a `HandleBase` type with mio.
+#[derive(Debug)]
+pub struct EventedHandle<T> where T: HandleBase {
+    /// The handle to be registered.
+    handle: T,
+
+    /// The current `Token` with which the handle is registered with mio.
+    token: Mutex<Option<Token>>,
+}
+
+impl<T> EventedHandle<T> where T: HandleBase {
+    /// Create a new `EventedHandle` which can be registered with mio
+    /// in order to receive event notifications.
+    pub fn new(handle: T) -> Self {
+        EventedHandle {
+            handle: handle,
+            token: Mutex::new(None),
+        }
+    }
+
+    /// Get a reference to the underlying `HandleBase`.
+    pub fn get_handle(&self) -> &T {
```

Comment:
@cramertj oh by this point it's mostly convention, but the thinking is that this is sort of a container rather than a conversion, and we use `get` for contains and `as` for conversions.